scripts/inactive_issue_checker.py

```diff
@@ -0,0 +1,492 @@
+
+        if response is None:
+            raise AssertionError(
+                'Received null res while fetching collaborators'
+            )
+        response.raise_for_status()
+
+        collaborators = set()
+        for collaborator_dict in response.json():
+            if isinstance(collaborator_dict, dict):
+                collaborators.add(collaborator_dict['login'])
+
+        return collaborators
+
+    def get_issue_events(self, issue: Issue) -> Optional[datetime.datetime]:
+        """Fetches and processes events for an issue.
+
+        Args:
+            issue: Issue. The issue to fetch events for.
+
+        Returns:
+            Optional[datetime.datetime]. The date of the latest event, if any.
+
+        Raises:
+            AssertionError. Raised if the response from the request is None.
+            requests.HTTPError. Raised if the request fails.
+        """
+        response = requests.get(
+            issue.events_url,
+            headers=self.rest_headers,
+            timeout=10
+        )
+        if response is None:
+            raise AssertionError('Received null res while fetching events')
+        response.raise_for_status()
+
+        events_dict = response.json()
+
+        if not events_dict:
+            return None
+
+        assignee_events = []
+        for event in events_dict:
+            if event.get('actor', {}).get('login') == issue.assignee_username:
+                assignee_events.append(event)
+
+        if not assignee_events:
+            return None
+
+        latest_event_date = max(
+            datetime.datetime.strptime(
+                event['created_at'],
+                '%Y-%m-%dT%H:%M:%SZ'
+            ).replace(tzinfo=datetime.timezone.utc)
+            for event in assignee_events
+        )
+        return latest_event_date
+
+    def get_issues_with_prs(self) -> Dict[int, Set[int]]:
+        """Fetches mapping of issues to their linked PRs using GraphQL.
+
+        Returns:
+            Dict[int, int]. Mapping of issue numbers to PR numbers.
+
+        Raises:
+            AssertionError. Raised if the response from the request is None.
+            requests.HTTPError. Raised if the request fails.
+        """
+        query = """
+        query($owner: String!, $name: String!, $cursor: String) {
+          repository(owner: $owner, name: $name) {
+            pullRequests(first: 100, states: [OPEN], after: $cursor) {
+              pageInfo {
+                hasNextPage
+                endCursor
+              }
+              nodes {
+                number
+                closingIssuesReferences(first: 1) {
+                  nodes {
+                    number
+                  }
+                }
+              }
+            }
+          }
+        }
+        """
+
+        req_credentials = {
+            'owner': self.repo_owner,
+            'name': self.repo_name,
+            'cursor': None
+        }
+
+        issue_to_prs: Dict[int, Set[int]] = {}
```

Comment:
Look into defining this as a Python collections.defaultdict(set) -- might be a cleaner way to do the aggregation.